### ISSUE TITLE
Leela Zero vs Leela?

### ISSUE THREAD

**Issue by luigio** [leela-zero/leela-zero#854]
Some noob's questions. I see on CGOS:

leela-0.11.0-p1600 - 2969 Elo
LZ-d6f3a6-t1-p1600 - 2876 Elo

What should we make of these numbers? If both Leela and Leela Zero ran the same number of playouts per unit of time, this would be a good estimation of their relative strengths, but that seems unlikely. Which one is faster and how many playouts does it run in the time it takes the slower one to run one playout? Also, how big is Leela's network compared to Leela Zero's?

My computer is not good enough to run a proper match between the two. Without a GPU, does the strength difference between Leela and Leela Zero increase or decrease?

Thanks!

**Comment by Dorus** [comment]
On my pc Leela 11 is easily 3-4x as fast as LeelaZ. Now this ratio might not hold for every pc, also my test was very dirty (didn't close down many background programs), but it should be clear Leela 11 is a lot faster.

**Comment by Dorus** [comment]
But also: LeelaZ by far does not max out my pc resources. I run 2 instances, so actually LeelaZ would be twice as fast if i used 2 threats. Also the next branch has a bunch of speedups. With those, LeelaZ might as well be equally fast as Leela 11. Who knows.

By numbers: LeelaZ gives 500 playouts/sec, Leela 11 gives 1800 nodes/sec.

**Comment by gcp** [comment]
Leela 0.11 uses Monte Carlo playouts, the speeds aren't comparable at all and comparing at a fixed number of playouts is somewhat meaningless.

**Comment by luigio** [comment]
Thanks. Still, it would be useful to know how playout speed compares between Leela 0.11 OpenCL and Leela Zero on the same computer. I'm assuming the Leela on CGOS is OpenCL (which doesn't work on my computer).

Dorus' comment is somewhat surprising. I think gcp mentioned once that Leela had a bigger network, so I thought it would be slower, not faster.

**Comment by gcp** [comment]
>I think gcp mentioned once that Leela had a bigger network, so I thought it would be slower, not faster.

Leela uses Monte Carlo playouts. The speeds aren't comparable at all, and comparing playouts is meaningless.

**Comment by luigio** [comment]
One can estimate strength difference with equal thinking time from data based on fixed number of playouts by knowing how long it takes each bot to run one playout, regardless of whether playouts are different in each bot. Of course, each bot takes a different amount of time to run one playout because playouts (and other stuff) are different in each. That's what makes it a useful piece of information.

**Comment by gcp** [comment]
Sure, but I am explaining you why the assumption that the network size would be related to the speed is not correct.

**Comment by jkiliani** [comment]
With increasing neural network size, Leela Zero is going to be increasingly GPU bound, and CPU usage is going to decrease further. Are there any plans to at some point incorporate the network evals of LZ into Leela, to gain the additional benefit of the Monte Carlo playouts? In https://github.com/gcp/leela-zero/issues/855 it is mentioned that this is already being done by other devs at the moment...

**Comment by gcp** [comment]
Mostly depends on me having time. At some point I'll retire from running this project and will probably update regular Leela with the relevant stuff from this project.

It will be interesting for example if the raw Zero network will be stronger than training regular Leela (with "smart" inputs) on the Zero data.

**Comment by roy7** [comment]
If at some point Leela Zero's mission of re-implementing Alpha Go Zero is "complete", the next step could be a fresh run with features. In theory it should be able to learn faster and get stronger, since if the features are truly useless it'll just generalize around them. If they aren't useless, the NN will learn to use the info optimally in time.

At that point the project would move from "copy AlphaGo Zero" to "create world's strongest GoO AI".

**Comment by jslechasseur** [comment]
It seems to me that leelaz is now much stronger than leela. Here's a 2H game I ran using the latest 0.13 version with the newest weights (85c6f2ad): http://eidogo.com/#z1ouHGp7

I used --noponder and time_settings 0 61 1. Both versions are running on my old GTX 680. The game looked grim of a while, but eventually leela slipped and missed this epic tesuji:
![image](https://user-images.githubusercontent.com/35378510/37889284-90d0e802-3099-11e8-9be4-1f2dcc732fc3.png)

The game went downhill from there for Leela...
